### DISCUSSION TITLE
Input.http - This value has no method

### DISCUSSION THREAD

**Discussion by Ceel0** [savonet/liquidsoap#3559] [User Support]
Hello,

I hope everyone is well.

I have a problem with my liquidsoap installation. I can't launch the radio.liq script.

Configuration : 
- Debian Bullseye 11
- Opam version 2.0.8
- Liquidsoap version 2.0.3-1
- ocurl version : 0.9.2

When i launch radio.liq i get : 
[input.http("myurl"),

Error 5: this value has no method http

How can i resolve my issue please ?

Thank you very much in advance.

Best regards,

Ceelo

**Comment by vitoyucepi** [comment]
Hi @Ceel0,
1. Did you install liquidsoap using opam?
   Have you tried using packages from the [release artifacts](https://github.com/savonet/liquidsoap/releases)?
1. Why did you install version `2.0.3` instead of `2.2.2` or at least `2.0.7`?
   Did you read a tutorial that recommends this exact version?
   Can you give me a link?
1. To use `input.http` you need to install `ffmpeg` from opam.
   Can you show me the result of `opam list`?

**Comment by Ceel0** [comment]
Hi @vitoyucepi,

Thank you for your reply.

You will find below my answers : 

Did you install liquidsoap using opam? YES

Have you tried using packages from the [release artifacts](https://github.com/savonet/liquidsoap/releases)? NO

Why did you install version 2.0.3 instead of 2.2.2 or at least 2.0.7? When i try an opam upgrade i get this message : 
"The following packages are not being upgraded because the new versions conflict with other installed packages:
  - camomile.2.0.0
  - cry.1.0.1
  - duppy.0.9.3
  - liquidsoap.2.2.2
  - menhir.20230608
  - menhirLib.20230608
  - menhirSdk.20230608
  - mm.0.8.4
  - pcre.7.5.0
However, you may "opam upgrade" these packages explicitly, which will ask permission to downgrade or uninstall the conflicting packages.
Nothing to do."

Did you read a tutorial that recommends this exact version? NO, it was installed by OPAM

Can you give me a link? N/A

To use input.http you need to install ffmpeg from opam.
Can you show me the result of opam list?
"debian@vps-9809a755:~$ opam list
# Packages matching: installed
# Name              # Installed # Synopsis
base-bigarray       base
base-threads        base
base-unix           base
camomile            1.0.2       A Unicode library
conf-fdkaac         1           Virtual package relying on fdk-aac
conf-ffmpeg         1           Virtual package relying on FFmpeg
conf-lame           1           Virtual package relying on lame
conf-libcurl        2           Virtual package relying on a libcurl system installation
conf-libpcre        1           Virtual package relying on a libpcre system installation
conf-mad            1           Virtual package relying on mad
conf-pkg-config     3           Check if pkg-config is installed and create an opam switch local pkgconfig folder
conf-taglib         1           Virtual package relying on taglib
conf-which          1           Virtual package relying on which
cry                 0.6.7       OCaml client for the various icecast & shoutcast source protocols
csexp               1.5.2       Parsing and printing of S-expressions in Canonical form
dtools              0.4.5       Library providing various helper functions to make daemons
dune                3.11.1      Fast, portable, and opinionated build system
dune-configurator   3.11.1      Helper library for gathering system configuration
duppy               0.9.2       Library providing monadic threads
fdkaac              0.3.3       Fraunhofer FDK AAC Codec Library
gen                 1.1         Iterators for OCaml, both restartable and consumable
lame                0.3.7       MP3 encoding library
liquidsoap          2.0.3-1     Swiss-army knife for multimedia streaming
mad                 0.5.2       Mad decoding library
menhir              20211128    An LR(1) parser generator
menhirLib           20211128    Runtime support library for parsers generated by Menhir
menhirSdk           20211128    Compile-time library for auxiliary tools related to Menhir
mm                  0.7.4       The mm library contains high-level to create and manipulate multimedia streams (audio, video, MIDI)
ocaml               4.11.1      The OCaml compiler (virtual package)
ocaml-compiler-libs v0.12.4     OCaml compiler libraries repackaged
ocaml-config        1           OCaml Switch Configuration
ocaml-system        4.11.1      The OCaml compiler (system version, from outside of opam)
ocamlfind           1.9.6       A library manager for OCaml
ocurl               0.9.2       Bindings to libcurl
pcre                7.4.6       Bindings to the Perl Compatibility Regular Expressions library
ppx_derivers        1.2.1       Shared [@@deriving] plugin registry
ppxlib              0.31.0      Standard infrastructure for ppx rewriters
sedlex              3.2         An OCaml lexer generator for Unicode
seq                 base        Compatibility package for OCaml's standard iterator type starting from 4.07.
sexplib0            v0.16.0     Library containing the definition of S-expressions and some base converters
stdlib-shims        0.3.0       Backport some of the new stdlib features to older compiler
taglib              0.3.10      Bindings for the taglib library"

Thank you very much for your help.

BR,

Ceelo

**Comment by Ceel0** [comment]
Hello @vitoyucepi ,

Sorry for the double post.

I installed "opam install ffmpeg" and it is working now.

Do you know how can i upgrade to liquidsoap 2.2.2 ?

BR,

Ceelo

**Reply by vitoyucepi** [reply]
You could upgrade to a new version of liquidsoap by
1. Creating a new ocaml installation.
   ```sh
   opam switch 4.14.1;
   eval $(opam env --switch=4.14.1);
   ```
   After that, you can install the latest available liquidsoap.

   <details>
   <summary>VM init script</summary>

   I use this script on a clean vm to test and debug opam installations, probably you'll find something useful.
   ```sh
   sudo apt-get update;
   sudo apt-get dist-upgrade;
   sudo apt-get install --no-install-recommends -y curl ca-certificates patch unzip bubblewrap git bzip2 gcc make libc6-dev g++;
   sudo curl -sS -L -o /usr/local/bin/opam https://github.com/ocaml/opam/releases/download/2.1.5/opam-2.1.5-x86_64-linux;
   sudo chmod +x /usr/local/bin/opam;
   opam init --bare -n;
   opam switch create 4.14.1;
   eval $(opam env --switch=4.14.1);
   opam install --confirm-level=unsafe-yes liquidsoap ffmpeg;
   ```

   </details>

1. Using the official container image [`savonet/liquidsoap:v2.2.2`](https://hub.docker.com/r/savonet/liquidsoap) or creating your own container image.
1. Installing from a [deb package](https://github.com/savonet/liquidsoap/releases), but the latest version requires *debian:12*.
